lib/hooks/userspace/hook-redis.js

```diff
@@ -32,6 +32,23 @@ function createClientWrap(createClient) {
   };
 }
 
+// Used for redis version > 2.3
+function createStreamWrap(create_stream) {
+  return function create_stream_trace() {
+    if (!this.stream) {
+      Object.defineProperty(this, 'stream', {
```

Comment:
Why does the define property need to happen at `create_stream` invocation? Could you define the proxy by wrapping the constructor?